octobot/community/authentication.py

```diff
@@ -201,19 +204,102 @@ async def login(self, email, password):
+
+    async def update_selected_device(self):
+        self.user_account.flush_device_details()
+        await self._load_device_if_selected()
+        if not self.user_account.has_selected_device_data():
+            self.logger.info("No selected device. Please select a device to enable your community features.")
+
+    async def _load_device_if_selected(self):
+        # 1. use user selected device id if any
+        if saved_uuid := self._get_saved_gql_device_id():
+            await self.select_device(saved_uuid)
+        else:
+            # 2. fetch all user devices and create one if none, otherwise if there is only one, use it
+            await self.load_user_devices()
+            if len(self.user_account.get_all_user_devices_raw_data()) == 0:
+                await self.select_device(
+                    self.user_account.get_device_id(
+                        self.create_new_device()
+                    )
+                )
+            # 3. fetch all user devices and if there is only one, use it
+            if len(self.user_account.get_all_user_devices_raw_data()) == 1:
+                await self.select_device(
+                    self.user_account.get_device_id(
+                        self.user_account.get_all_user_devices_raw_data()[0]
+                    )
+                )
+            # more than one possible device, can't auto-select one
+
+    async def select_device(self, device_id):
+        self.user_account.set_selected_device_raw_data(await self.fetch_device(device_id))
+        self.user_account.gql_device_id = device_id
+        self._save_gql_device_id(self.user_account.gql_device_id)
+        await self.on_new_device_select()
+
+    async def load_user_devices(self):
+        self.user_account.set_all_user_devices_raw_data(await self._fetch_devices())
+
+    async def on_new_device_select(self):
+        await self._update_feed_device_uuid()
+
+    async def _fetch_devices(self):
+        query, variables = graphql_requests.select_devices(self.user_account.gql_user_id)
+        select_resp = await self.async_graphql_query(query, variables=variables)
+        if select_resp.status == 200:
+            json_resp = await select_resp.json()
+            return json_resp["data"]["devices"]
+        else:
+            raise RuntimeError(f"Error when fetching devices, code: {select_resp.status}, "
+                               f"text: {await select_resp.text()}")
+
+    async def fetch_device(self, device_id):
+        query, variables = graphql_requests.select_device(device_id)
+        select_resp = await self.async_graphql_query(query, variables=variables)
+        if select_resp.status == 200:
+            json_resp = await select_resp.json()
+            return json_resp["data"]["device"]
+        else:
+            raise RuntimeError(f"Error when fetching device, code: {select_resp.status}, "
+                               f"text: {await select_resp.text()}")
```

Comment:
Should we wrap this call? Because we already duplicate multiple times `async_graphql_query` with http code check and `await select_resp.json()`?